crates/crypto/bls/src/supranational/signature.rs

```
@@ -0,0 +1,145 @@
+        Ok(Self { inner: signature })
+    }
+}
+
+impl Serialize for BlsSignature {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let val = hex::encode(self.inner);
+        serializer.serialize_str(&val)
+    }
+}
+
+impl<'de> Deserialize<'de> for BlsSignature {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let result: String = Deserialize::deserialize(deserializer)?;
+        let result = hex::decode(&result).map_err(serde::de::Error::custom)?;
+        let mut signature = [0u8; 96];
+        signature.copy_from_slice(&result);
+        Ok(Self { inner: signature })
+    }
+}
+
+impl TreeHash for BlsSignature {
+    fn tree_hash_type() -> tree_hash::TreeHashType {
+        TreeHashType::Vector
+    }
+
+    fn tree_hash_packed_encoding(&self) -> PackedEncoding {
+        PackedEncoding::from_vec(self.inner.to_vec())
+    }
+
+    fn tree_hash_packing_factor() -> usize {
+        1
+    }
+
+    fn tree_hash_root(&self) -> Hash256 {
+        merkle_root(&self.inner, 1)
+    }
+}
+
+impl BlsSignature {
+    fn to_blst_signature(&self) -> anyhow::Result<Signature> {
+        Signature::from_bytes(&self.inner)
+            .map_err(|e| anyhow::anyhow!("Failed to convert signature: {:?}", e))
+    }
+
+    /// Verifies a BLS signature against a public key and message.
+    ///
+    /// This function will return `false` in any error case, including:
+    /// - If the signature bytes are invalid or malformed
+    /// - If the public key bytes are invalid or malformed
+    /// - If the actual signature verification fails
+    ///
+    /// # Arguments
+    /// * `pubkey` - The public key to verify against
+    /// * `message` - The message that was signed
+    ///
+    /// # Returns
+    /// `bool` - true if the signature is valid, false otherwise
+    pub fn verify(&self, pubkey: &PubKey, message: &[u8]) -> bool {
```

We shouldn't hide the error using an `is_ok_and` we should return it